server/src/main/java/org/elasticsearch/search/sort/SortValue.java

```diff
@@ -338,4 +346,69 @@ public Number numberValue() {
             return Double.NaN;
         }
     }
+
+    private static class EmptySortValue extends SortValue {
+
+        public static final String NAME = "empty";
+        private final byte dummyKey;
+
+        private EmptySortValue() {
+            this.dummyKey = (byte) 0;
+        }
+
+        @Override
+        public String getWriteableName() {
+            return NAME;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeByte(dummyKey);
+        }
+
+        @Override
+        public Object getKey() {
+            return dummyKey;
+        }
+
+        @Override
+        public String format(DocValueFormat format) {
+            return format.format(dummyKey).toString();
```

Comment:
I'd just return `null` or "" or something here. I'm not sure what is right, but `0` probably isn't.